src/ActionOpen.java

```diff
@@ -0,0 +1,81 @@
+import java.io.File;
+import java.io.FileReader;
+
+import javax.swing.JFileChooser;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JTextPane;
+
+/*
+ * Class that handles the Open Action
+ */
+public class ActionOpen extends Action {
+
+	// Handles opening of the file
+	public Integer doAction(SimpleNotePad simpleNotePad, JTextPane notePad, Integer count, JMenuItem recentFile,
+			JMenuItem fileMenu, JMenuItem[] recent) {
+
+		JFileChooser fileChooser = new JFileChooser();
+		fileChooser.setCurrentDirectory(new File(System.getProperty("user.home"))); // sets home location for when file
+																					// is opened
+		int result = fileChooser.showOpenDialog(simpleNotePad);
+		if (result == JFileChooser.APPROVE_OPTION) {
+			File selectedFile = fileChooser.getSelectedFile();
+			String path = selectedFile.toString();
+			openFile(path, notePad);
+
+			// If there are more then 3 items then the recent array
+			if (count >= 3) {
+				recentFile.remove(recent[0]); // removes top recent item
+				recent[0] = recent[1]; // shifts the next recent item to top
+				recent[1] = recent[2]; // shifts the last recent item to middle
+				recent[2] = new JMenuItem(path); // creates new menu item for new recent file
+
```

Comment:
Using an JMenuItem array is fine; however, there may be cleaner approaches. We should consider using an ArrayList<JMenuItem> instead.

Example snippet
```suggestion
	ArrayList<JMenuItem> recent;
	JMenuItem recentMenuItem = new JMenuItem(path);
		if(recent.contains(recentMenuItem))
			recent.remove(recentMenuItem);
		if(recent.size() >= 5)
			recent.remove(0);
		recent.add(recentMenuItem);
```

This way, we don't have to hard-code the index to perform a shift operation.